server/service/src/ledger_fix/find_ledger_discrepancies.rs

```diff
@@ -0,0 +1,218 @@
+use repository::{
+    stock_line_ledger_discrepancy::{
+        StockLineLedgerDiscrepancyFilter, StockLineLedgerDiscrepancyRepository,
+    },
+    EqualFilter, RepositoryError, StockLineFilter, StorageConnection,
+};
+use thiserror::Error;
+
+use crate::sync::{ActiveStoresOnSite, GetActiveStoresOnSiteError};
+
+#[derive(Error, Debug)]
+pub enum FindStockLineLedgerDiscrepanciesError {
+    #[error(transparent)]
+    DatabaseError(#[from] RepositoryError),
+    #[error("Failed to get active stores on site")]
+    GetActiveStoresOnSiteError(#[from] GetActiveStoresOnSiteError),
+}
+
+pub(super) fn find_stock_line_ledger_discrepancies(
+    connection: &StorageConnection,
+) -> Result</* stock line ids */ Vec<String>, FindStockLineLedgerDiscrepanciesError> {
+    let active_stores = ActiveStoresOnSite::get(connection)?;
+
+    let filter = StockLineLedgerDiscrepancyFilter {
+        stock_line: Some(
+            StockLineFilter::new()
+                .store_id(EqualFilter::equal_any_or_null(active_stores.store_ids())),
+        ),
+    };
+
+    Ok(StockLineLedgerDiscrepancyRepository::new(connection)
+        .query(Some(filter))?
+        .into_iter()
+        .map(|r| r.id)
+        .collect())
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::test_helpers::{setup_all_with_data_and_service_provider, ServiceTestContext};
+    use chrono::Utc;
+    use repository::{
+        mock::{mock_item_a, mock_name_a, mock_store_a, mock_store_b, MockData, MockDataInserts},
+        InvoiceLineRow, InvoiceLineType, InvoiceRow, InvoiceStatus, InvoiceType,
+        KeyValueStoreRepository, StockLineRow,
+    };
+    use util::inline_edit;
+
+    fn make_movements(stock_line: StockLineRow, date_quantity: Vec<(i64, i64)>) -> MockData {
```

Comment:
quite happy with this, might be useful in other places @Chris-Petty would be useful for you in doing tests mocks